service/src/worker.rs

```diff
@@ -89,12 +92,7 @@ where
 	}
 
 	fn peers(&self) -> WorkerResult<Vec<EnclaveMetadata>> {
-		let mut peers = self.node_api.all_enclaves()?;
-		peers.retain(|e| {
-			e.url.trim_start_matches("ws://").trim_start_matches("wss://")
-				!= self.config.worker_url()
-		});
-		Ok(peers)
+		Ok(self.node_api.all_enclaves()?)
```

Comment:
Send block to oneself as well

Reply:
I am not sure if we should send it to ourselves, now we even check upon block import if we were the author and handle it differently. This would be simpler if this weren't the case.

But I also see that it requires extra complexity at the block-production side. Have you been weighing these two options?

Reply:
On second thought, I think your solution might be better.

Reply:
I'm unsure about your comment. Without sending it to ourselves, how should be import the block?

The check if we are the author is only for top_pool_call removal, import is happening the same way as for non-block-authors.

Reply:
I agree with @haerdib on this one, the block importing is done completely agnostic of whether it's our own block that we import. The only check we have, is if we need to remove the call from the top pool. There we had the two options:
1. Always attempt to remove the call from the top pool when we import a block (and will only succeed for the worker which produced the block)
2. Check if it's our own block that we imported, so only the 'correct' worker removes the call from the top-pool.

Reply:
Yes, I agree. @haerdib's approach is better.